xio-core/src/main/java/com/xjeffrose/xio/http/ClientFactory.java

```diff
@@ -3,26 +3,34 @@
 import com.xjeffrose.xio.bootstrap.ChannelConfiguration;
 import com.xjeffrose.xio.bootstrap.ClientChannelConfiguration;
 import com.xjeffrose.xio.client.ClientConfig;
+import com.xjeffrose.xio.tracing.XioTracing;
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.util.AttributeKey;
 
 // TODO(CK): only use case is currently for proxy clients, establish more use cases.
 /** Base class for generating http clients */
 public abstract class ClientFactory {
 
+  protected XioTracing tracing;
+
   private static final AttributeKey<Client> CLIENT_KEY =
       AttributeKey.newInstance("xio_http_client_key");
 
+  public ClientFactory(XioTracing tracing) {
+    this.tracing = tracing;
+  }
+
   public ClientChannelConfiguration channelConfig(ChannelHandlerContext ctx) {
     return ChannelConfiguration.clientConfig(ctx.channel().eventLoop());
   }
 
-  public abstract Client createClient(ChannelHandlerContext ctx, ClientConfig config);
+  public abstract Client createClient(
+      ChannelHandlerContext ctx, ClientConfig config, XioTracing tracing);
```

Comment:
lose `tracing` here as a parameter and have the `@Override` method get at it through the getter

Reply:
fixed